code/datums/components/rot.dm

```diff
@@ -46,14 +72,34 @@
 		return
 
 	// No decay if formaldehyde in corpse or when the corpse is charred
-	if(C.reagents.has_reagent("formaldehyde", 15) || C.has_trait(TRAIT_HUSK))
+	if(C.reagents.has_reagent("formaldehyde", 15))
 		return
 
 	// Also no decay if corpse chilled or not organic/undead
-	if(C.bodytemperature <= T0C-10 || (!(MOB_ORGANIC in C.mob_biotypes) && !(MOB_UNDEAD in C.mob_biotypes)))
+	if(C.bodytemperature <= T0C-10)
 		return
 
 	..()
 
 /datum/component/rot/gibs
 	amount = MIASMA_GIBS_MOLES
+
+/datum/component/rot/bodypart
+	amount = MIASMA_GIBS_MOLES
+
+/datum/component/rot/bodypart/process()
+	var/obj/item/bodypart/BP = parent
+	if(BP.owner || BP.burn_dam > 100)
```

Comment:
Instead of checking in process make it get removed in the process of being reattached/burning up/whatever event makes this get removed. Using process to check for changes in state is a bad idea